ui/src/components/app-form/app-form.tsx

```diff
@@ -65,11 +71,192 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
+  const [error, setError] = useState<string | null>(null); // Store validation errors
+  const [shouldValidate, setShouldValidate] = useState(false); // To control validation trigger
+  const repoUrlRef = useRef<HTMLInputElement>(null); // Ref for Git Repository URL field
+  const {
+    control,
+    handleSubmit,
+    setFocus,
+    setValue,
+    reset,
+    watch,
+    formState: { errors, isDirty, isValid },
+  } = useForm<AppFormInput>({
+    defaultValues: {
+      display_name: '',
+      description: '',
+      framework: '',
+      thumbnail: '',
+      filepath: initialFilepath,
+      conda_env: '',
+      custom_command: '',
+      profile: '',
+      is_public: false,
+      keep_alive: false,
+      share_with: {
+        users: [],
+        groups: [],
+      },
+    },
+  });
+
+  const initialValues = {
+    display_name: '',
+    description: '',
+    framework: '',
+    thumbnail: '',
+    filepath: initialFilepath,
+    conda_env: '',
+    custom_command: '',
+    profile: '',
+    is_public: false,
+    keep_alive: false,
+    share_with: {
+      users: [],
+      groups: [],
+    },
+  };
+
+  const watchedFields = watch();
+  const hasChanges =
+    JSON.stringify(watchedFields) !== JSON.stringify(initialValues);
+
+  useEffect(() => {
+    // Reset form fields when deployOption changes
+    reset({
+      display_name: '',
+      description: '',
+      framework: '',
+      thumbnail: '',
+      filepath: '',
+      conda_env: '',
+      custom_command: '',
+      profile: '',
+      is_public: false,
+      keep_alive: false,
+    });
+
+    // Reset additional state if necessary
+    setDescription('');
+    setRepoData(null);
+    setIsUrlValid(false);
+  }, [deployOption, reset]);
+
+  useEffect(() => {
+    const filepathFromQuery = decodeURIComponent(
+      searchParams.get('filepath') || '',
+    );
+    setValue('filepath', filepathFromQuery); // Update the form field value
+  }, [searchParams, setValue]);
+
+  const labelStyle = {
+    fontSize: '1rem',
+    transform: 'translate(14px, -6px) scale(0.75)',
+    color: 'rgba(0, 0, 0, 0.87)', // Adjust color based on your needs
+    top: '-2px',
+    left: '-5px',
+    padding: '0 4px',
+    zIndex: 1,
+    backgroundColor: '#fafbfc', // Set white background
+    paddingRight: '4px',
+    paddingLeft: '4px',
+  };
+  // Validate the URL format and fetch repo data
+  const validateGitUrl = async () => {
+    // Trigger validation after the button is clicked
+    setShouldValidate(true);
+
+    const gitRepoUrlPattern =
+      /^(https?:\/\/)?[\w.-]+\/[\w-]+\/[\w-]+(\.git)?$/i;
+
+    if (!gitRepoUrlPattern.test(gitUrl)) {
+      setError('Invalid Git repository URL');
+      setIsUrlValid(false);
+      setOpenModal(true);
+      return;
+    }
+
+    try {
+      setIsFetching(true);
+      setError(null);
+
+      const response = await axios.post('/app-config-from-git/', {
+        url: gitUrl,
```

Comment:
This also needs to include `config_directory` (conda project path) and `ref` (branch) in the request.

See: https://github.com/nebari-dev/jhub-apps/blob/f97e8bde77287c81719239688c0f179a0681fa07/jhub_apps/service/models.py#L58-L63

Reply:
Included the other key:values; config_directory and ref